### DISCUSSION TITLE
Difference between  OT_RADIO_CAPS_TRANSMIT_RETRIES  & OT_RADIO_CAPS_CSMA_BACKOFF Radio capability

### DISCUSSION THREAD

**Discussion by AhmedEsmailm** [openthread/openthread#9676] [Q&A] [answered]
Hello OT team,

The below description for  OT_RADIO_CAPS_TRANSMIT_RETRIES  & OT_RADIO_CAPS_CSMA_BACKOFF capabilities are mentioned in API reference and Radio .h 
OT_RADIO_CAPS_CSMA_BACKOFF | Radio supports CSMA backoff for frame transmission (but no retry).
OT_RADIO_CAPS_TRANSMIT_RETRIES | Radio supports tx retry logic with collision avoidance (CSMA).

I understand  from this description that OT_RADIO_CAPS_CSMA_BACKOFF  should be set for the radio driver  that  supports backoff  but doesn't implement the CSMA retry. while OT_RADIO_CAPS_TRANSMIT_RETRIES  should be set for radio  driver  that implements CSMA retries. Based on that, only of these capabilities should be set for certain radio driver. i.e if OT_RADIO_CAPS_TRANSMIT_RETRIES   is supported then this implies that OT_RADIO_CAPS_CSMA_BACKOFF  is supported as part on it and the OT_RADIO_CAPS_CSMA_BACKOFF   shouldn't be set.

Could you please  confirm my understanding

Thanks in advance,
Ahmed Ismail

</body>
</html>

</body>
</html>

Thanks,
Ahmed Ismail

**Comment by abtink** [comment, marked as answer]
A radio supporting `OT_RADIO_CAPS_TRANSMIT_RETRIES` should also already support `OT_RADIO_CAPS_CSMA_BACKOFF` (and  `OT_RADIO_CAPS_ACK_TIMEOUT`).

I recommend including all these capabilities in the `otPlatRadioGetCaps()` return value for such a case. This ensures that `GetCaps()` accurately represents all supported capabilities. 

The OT stack may not explicitly check for `OT_RADIO_CAPS_CSMA_BACKOFF` if `OT_RADIO_CAPS_TRANSMIT_RETRIES` is present, but it won't be a good idea to assume this in radio platform implementation (and exclude the `OT_RADIO_CAPS_CSMA_BACKOFF` in `GetCaps()`).

**Comment by AhmedEsmailm** [comment]
Thanks for the info
